root/static/scripts/common/components/Relationship.js

```diff
@@ -0,0 +1,178 @@
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import linkedEntities from '../linkedEntities';
+import {commaOnlyListText} from '../i18n/commaOnlyList';
+import {bracketedText} from '../utility/bracketed';
+import formatDate from '../utility/formatDate';
+import isDateEmpty from '../utility/isDateEmpty';
+import {interpolate, getExtraAttributes} from '../../edit/utility/linkPhrase';
+import type {LinkPhraseProp} from '../../edit/utility/linkPhrase';
+
+import DescriptiveLink from './DescriptiveLink';
+
+type Props = {
+  +phraseField?: LinkPhraseProp,
+  +relationship: RelationshipT,
+};
+
+const RelationshipDates = ({relationship}: {relationship: RelationshipT}) => {
+  const beginDate = relationship.begin_date;
+  const endDate = relationship.end_date;
+  const formattedBeginDate = formatDate(beginDate);
+  const formattedEndDate = formatDate(endDate);
+  const hasBeginDate = !isDateEmpty(beginDate);
+  const hasEndDate = !isDateEmpty(endDate);
+  const hasBeginDay = beginDate && beginDate.day;
+
+  return (
+    hasBeginDate ? (
+      <>
+        {' '}
+        {hasEndDate ? (
+          formattedBeginDate === formattedEndDate ? (
+            hasBeginDay
+              ? exp.l('on {date}', {date: formattedBeginDate})
+              : exp.l('in {date}', {date: formattedBeginDate})
+          ) : (
+            exp.l(
+              'from {begin_date} until {end_date}',
+              {
+                begin_date: formattedBeginDate,
+                end_date: formattedEndDate,
+              },
+            )
+          )
+        ) : (
+          relationship.ended
+            ? exp.l('from {date} to ????', {date: formattedBeginDate})
+            : exp.l('from {date} to present', {date: formattedBeginDate})
+        )}
+      </>
+    ) : (
+      hasEndDate ? (
+        <>
+          {' '}
+          {exp.l('until {date}', {date: formattedEndDate})}
+        </>
+      ) : (
+        relationship.ended ? (
+          <>
+            {' '}
+            {bracketedText(l('ended'))}
+          </>
+        ) : null
+      )
+    )
+  );
+};
+
+const HistoricRelationshipContent = ({
+  relationship,
+}: {relationship: RelationshipT}) => {
+  const linkType = linkedEntities.link_type[relationship.linkTypeID];
+  const extraAttributes = getExtraAttributes(
+    relationship,
+    'link_phrase',
+    false, /* forGrouping */
+  );
+  return (
+    <>
+      {linkType.link_phrase /* hardcoded untranslatable historical hack */}
+      {' '}
+      <DescriptiveLink
+        // Used for historic edits which hardcode entity1 as target
+        content={relationship.entity1_credit}
+        entity={relationship.target}
+      />
+      <RelationshipDates relationship={relationship} />
+      {extraAttributes ? (
+        <>
+          {' '}
+          {bracketedText(commaOnlyListText(extraAttributes.map(
+            attribute => attribute.typeName,
+          )))}
```

Comment:
Looks like this should use root/static/scripts/common/utility/displayLinkAttribute.js

Reply:
I had made that change (well, displayLinkAttributes because I see no issue with it being commaList instead) in the relationship edit PR, but I guess I can move it to here already.